packages/web-components/fast-element/src/components/fast-definitions.ts

```diff
@@ -220,15 +221,21 @@ export class FASTElementDefinition<
         type: TType,
         nameOrDef?: string | PartialFASTElementDefinition
     ): FASTElementDefinition<TType> {
-        const found = fastElementRegistry.getByType(type);
-
-        if (found) {
+        if (fastElementBaseTypes.has(type) || fastElementRegistry.getByType(type)) {
             return new FASTElementDefinition<TType>(class extends type {}, nameOrDef);
         }
 
         return new FASTElementDefinition<TType>(type, nameOrDef);
     }
 
+    /**
+     * Registers a FASTElement base type.
+     * @param type - The type to register as a base type.
+     */
+    public static registerBaseType(type: Function) {
```

Comment:
Is there a use for this static for consuming authors, or should we mark it internal to narrow the API surface?

Reply:
Good call. I think we can mark this internal.